.github/workflows/ci.yaml

```
@@ -0,0 +1,63 @@
+name: Pipeline
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

There seems to be a newer version available. Could you please also check the other actions?
```suggestion
        uses: actions/checkout@v4
```